awx/ui/client/src/instance-groups/container-groups/edit-container-group.controller.js

```diff
@@ -0,0 +1,72 @@
+function EditContainerGroupController($rootScope, $scope, $state, models, strings, i18n, EditContainerGroupDataset) {
+
+  const vm = this || {};
+  const {
+    instanceGroup,
+    credential
+  } = models;
+  $rootScope.breadcrumb.instance_group_name = instanceGroup.get('name');
+
+  vm.mode = 'add';
+  vm.strings = strings;
+  vm.panelTitle = EditContainerGroupDataset.data.name;
+  vm.lookUpTitle = strings.get('container.LOOK_UP_TITLE');
+
+  vm.form = instanceGroup.createFormSchema('post');
+  vm.form.name.required = true;
+  vm.form.credential = {
+    type: 'field',
+    label: i18n._('Credential'),
+    id: 'credential'
+  };
+  vm.form.credential._resource = 'credential';
+  vm.form.credential._route = "instanceGroups.editContainerGroup.credentials";
+  vm.form.credential._model = credential;
+  vm.form.credential._displayValue = EditContainerGroupDataset.data.summary_fields.credential.name;
+  vm.form.credential.required = true;
+  vm.form.credential._value = EditContainerGroupDataset.data.summary_fields.credential.id;
+  vm.podSpec = {
+    type: 'textarea',
+    id: 'pod_spec'
+  };
+
+  vm.podSpec.label = strings.get('container.POD_SPEC_LABEL');
+  vm.form.extraVars = extraVarsDetails();
+  vm.extraVars = {
+    type: 'textarea',
+    id: 'extraVars'
+  };
+  vm.form.extraVars = extraVarsDetails();
```

Comment:
assuming one of these `vm.form.extraVars` assignments is unnecessary, we don't need a function for `extraVarsDetails`

instead of:

```js
vm.form.extraVars = extraVarsDetails();
```

we can do:

```js
vm.form.extraVars = {
    label: strings.get('container.POD_SPEC_LABEL'),
    value: EditContainerGroupDataset.data.pod_spec_override,
    name: 'extraVars'
};
```